src/components/datepicker/DatePickerInput.tsx

```diff
@@ -0,0 +1,197 @@
+  /** Object with props for the Input component. */
+  inputProps: InputProps;
+  /** If true, component will be rendered in inverse mode. */
+  inverse: boolean;
+  /** The language ISO locale code ('en-GB', 'nl-BE', 'fr-FR',...). */
+  locale: DateTimeFormatOptions | string;
+  /** Callback function that is fired when the date has changed. */
+  onChange: () => void;
+  /** Callback function that is fired when the popover with the calendar gets closed (unfocused) */
+  onBlur: () => void;
+  /** Object with props for the Popover component. */
+  popoverProps: object;
+  /** The current selected date. */
+  selectedDate: Date;
+  /** Size of the Input & DatePicker components. */
+  size: 'small' | 'medium' | 'large';
+  /** Overridable size of the Input component. */
+  inputSize: 'small' | 'medium' | 'large';
+  /** Overridable size of the DatePicker component. */
+  datePickerSize: 'small' | 'medium' | 'large';
+  /** Whether the picker should automatically open on input focus. True by default. */
+  openPickerOnFocus: boolean;
+}
+
+interface DayPickerProps {
+  numberOfMonths: number;
+  showOutsideDays: boolean;
+  showWeekNumbers: boolean;
+  withMonthPicker: boolean;
+}
+
+const DatePickerInput = ({
+  className,
+  dayPickerProps,
+  footer,
+  inverse = false,
+  inputProps,
+  locale = 'en-GB',
+  popoverProps,
+  size = 'medium',
+  inputSize,
+  datePickerSize,
+  formatDate: customFormatDate,
+  openPickerOnFocus = true,
+  onChange,
+  onBlur,
+  ...others
+}: DatePickerInputProps) => {
+  const [isPopoverActive, setIsPopoverActive] = useState(false);
+  const [popoverAnchorEl, setPopoverAnchorEl] = useState<EventTarget | null>(null);
+  const [selectedDate, setSelectedDate] = useState<Date | null>(null);
+
+  useEffect(() => {
+    if (others.selectedDate !== undefined && others.selectedDate !== selectedDate) {
+      setSelectedDate(others.selectedDate);
+      return;
+    }
+
+    setSelectedDate(null);
+  }, [others.selectedDate]);
+
+  const getFormattedDate = () => {
+    if (!selectedDate) {
+      return '';
+    }
+
+    if (!customFormatDate) {
+      return formatDate(selectedDate, locale as DateTimeFormatOptions);
+    }
+
+    return customFormatDate(selectedDate, locale);
+  };
+
+  const handleInputFocus = (event: React.FocusEvent<HTMLInputElement>) => {
+    if (inputProps.readOnly) {
+      return;
+    }
+
+    if (openPickerOnFocus) {
+      setPopoverAnchorEl(event.currentTarget);
+      setIsPopoverActive(true);
+      inputProps.onFocus && inputProps.onFocus(event);
+    } else {
+      inputProps.onFocus && inputProps.onFocus(event);
+    }
+  };
+
+  const handleInputClick = (event: React.MouseEvent<HTMLInputElement> | React.FocusEvent<HTMLInputElement>) => {
+    if (!openPickerOnFocus) {
+      setPopoverAnchorEl(event.currentTarget);
+      setIsPopoverActive(true);
+      inputProps.onFocus && inputProps.onFocus(event as React.FocusEvent<HTMLInputElement>);
```

Comment:
Don't think we should to this "casting", cause you cast it to something it isn't.
I guess it complains now because you're using the InputProps from react-select.
If we use our own, we can actually change the onFocus handler types, and use `event: React.MouseEvent<HTMLInputElement> | React.FocusEvent<HTMLInputElement>`